### PR TITLE
particle/contiguous/view interop

### PR DESCRIPTION
**Pull request by Dekken**
ref: https://github.com/PHAREHUB/PHARE/issues/295
todo: 
- benchmark loop methods for ContiguousParticles

### CODE REVIEW

File: tests/core/data/particles/CMakeLists.txt [hunk 1]
```diff
@@ -2,19 +2,38 @@ cmake_minimum_required (VERSION 3.9)
...
 
-add_no_mpi_phare_test(${PROJECT_NAME} ${CMAKE_CURRENT_BINARY_DIR})
+  add_phare_test(${test_name} ${CMAKE_CURRENT_BINARY_DIR})
+
+endfunction(_particles_test)
+
+_particles_test(test_main.cpp test-particles)
+_particles_test(test_interop.cpp test-particles-interop)
+
+# set(SOURCES test_main.cpp)
```
Comment (nicolasaunai) [inline tests/core/data/particles/CMakeLists.txt:24]: remove comments ?

File: tests/core/data/particles/test_interop.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                particle.delta  = ConstArray<float, dim>(i + 1);
+                particle.v      = ConstArray<double, 3>(particle.weight + 2);
+                i++;
+            }
+        }
+
+        for (auto const& particle : particles)
+        {
+            auto i = particleArray.size();
+            EXPECT_EQ(particles.weight[i], i + 1); // fastest
```
Comment (nicolasaunai) [inline tests/core/data/particles/test_interop.cpp:79]: typo ? how could that work?

**Review by nicolasaunai — APPROVED** [review]

**Comment by nicolasaunai** [comment]
to check why it fails before merging I guess

**Comment by Dekken** [comment]
> to check why it fails before merging I guess

can't see any reason why, @jeandet suggested it might be jave heap size related,

**Comment by Dekken** [comment]
> to check why it fails before merging I guess

https://github.com/PHAREHUB/PHARE/issues/327

**Comment by Dekken** [comment]
GPU branch is gonna conflict so hard

**Comment by Dekken** [comment]
> GPU branch is gonna conflict so hard

ouch 
image

**Pull request merged**